### PR TITLE
DASH-31 Add search admin logic

### PR DESCRIPTION
**Pull request by NikolasStelmakh**

### CODE REVIEW

File: server/src/services/adminUsers/Search.mjs [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+
+import Base from '../Base';
+import { Order, User } from '../../models';
+import { ADMIN_ROLES_IDS } from '../../config/constants.js';
+import X from '../Exception.mjs';
+
+const { Op } = sequelize;
+
+const validateEmail = email => {
+  return email.match(
```
Comment (andreyvEze) [inline server/src/services/adminUsers/Search.mjs:12]: there should be option in the current validation, please check and use it

File: client/src/components/MainHeader/components/AdminSearch.tsx [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+  const handleSearch = () => {
+    dispatch(adminSearch(searchValue));
+    if (location.pathname !== ROUTES.ADMIN_SEARCH_RESULT)
+      history.push(ROUTES.ADMIN_SEARCH_RESULT);
+  };
+
+  return (
+    <>
+      {isAdmin ? (
```
Comment (andreyvEze) [inline client/src/components/MainHeader/components/AdminSearch.tsx:26]: could we move it to `LoggedActionButtons` component ?

File: server/src/services/adminUsers/Search.mjs [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+      //
+    }
+
+    const result = {
+      users: [],
+      orders: [],
+    };
+
+    if (isEmail) {
+      // could be useful in future
```
Comment (andreyvEze) [inline server/src/services/adminUsers/Search.mjs:69]: create in this class `userSearch method and move all this commented logic there`

**Review by andreyvEze — CHANGES_REQUESTED** [review]
check comments

File: client/src/components/MainHeader/components/AdminSearch.tsx (hunk 2 above)
Comment (NikolasStelmakh) [reply client/src/components/MainHeader/components/AdminSearch.tsx:26]: why?)

**Pull request merged**